src/lib/bootloader/stage1_device.rb

```diff
@@ -57,24 +56,29 @@ def underlaying_devices_for(dev)
       res.uniq
     end
 
+    def pv_disk(vgs)
+      pvs = usable_pvs(vgs)
+      pvs.map { |p| p.ancestors.find { |a| a.is?(:disk) }.name }
+    end
+
     # get one level of underlaying devices, so no recursion deeper
     def underlaying_devices_one_level(dev)
       # FIXME: there is nothing like this right now in libstorage-ng
       #  a_vg.name #=> "/dev/vgname"
       # revisit when such thing exist
       match = /^\/dev\/(\w+)$/.match(dev)
       if match && match[1]
-        vgs = devicegraph.volume_groups.with(vg_name: match[1])
-        return usable_pvs(vgs).disks.map(&:name) unless vgs.empty?
+        vgs = devicegraph.lvm_vgs.select { |v| v.vg_name == match[1] }
+        return pv_disk(vgs) unless vgs.empty?
       end
 
       # FIXME: there is nothing like this right now in libstorage-ng
       #  a_lv.name #=> "/dev/vgname/lvname"
       # revisit when such thing exist
       match = /^\/dev\/\w+\/(\w+)$/.match(dev)
       if match && match[1]
-        lvs = devicegraph.logical_volumes.with(lv_name: match[1])
-        return usable_pvs(lvs.vgs).map { |pv| pv.blk_device.name } unless lvs.empty?
+        vgs = devicegraph.lvm_lvs.select { |v| v.lv_name == match[1] }.map(&:lvm_vg)
+        return usable_pvs(vgs).map { |v| v.blk_device.name } unless vgs.empty?
```

Comment:
Same than above. We probably don't need to work with lists anymore, so this could be:
```ruby
vg = devicegraph.lvm_lvs.detect { |v| v.name == dev }.lvm_vg
return usable_pvs(vg).map { |v| v.blk_device.name } if vg
```